An unfinished mod selector

ScaredKurufinve opened the pull request.

ScaredKurufinve committed:
Added a publicization for Owlcat.UI
Please, fix my ugly formatting.

ScaredKurufinve committed:
Changed the access modifier to an overriden method due to the publicization

ScaredKurufinve committed:
Changed the access modifier to an overriden method due to the publicization

ScaredKurufinve committed:
Changed the access modifier to an overriden method due to the publicization

ScaredKurufinve committed:
Added an empty Localized string to be used on several occasions

ScaredKurufinve committed:
New wrapper ckass which contains UI Setting Groups and mod info to generate the description which will be displayed when you hover over the entry in the dropdown. Info may be auto-generate out of a Mod Entry or set manually.

ScaredKurufinve committed:
New Setting Entity class with a ModsmenuEntry as a possible setting - on value changed redraws the setting screen to display settings of the chosen mod.

ScaredKurufinve committed:
Changed a postfix of the Switch Settings Screen to a boolean Prefix returning false. I intended to add a toggle setting allowing to disable the mod at run-time (there's a field in mod info for that), but then realized it won't be possible to properly enable it back after reloading.

ScaredKurufinve committed:
New UI Settings Dropdown - the main feature of the PR. I did not add any new VM and expect it to work correctly with the default dropdown.

ScaredKurufinve committed:
Updated the Builder to add new functionality.
ATTENTION - Build() function gotthe signature changed, it now returns generated mod info among other things

ScaredKurufinve committed:
Changed the AddSettings functions to fit the new reality

ScaredKurufinve committed:
added a local variable to store the modEntry so that it can be used to auto-generate ModInfo in TestSettings

ScaredKurufinve committed:
Changed the whole AddSettings stuff to fit the new reality

ScaredKurufinve committed:
Added new methods to Test Settings.
In particualar, concatenate test settings and extra settings into a single mod entry thhrough AddAnotherGroup

ScaredKurufinve committed:
Some changes. Realized I was giving the mod description instead of the mod name to the dropdown. Moved the call to redraw the screen from the setting itself to the UISetting. Deleted a duplicated call to add a ModEntry to the list of entries. Added a bunch of summaries

ScaredKurufinve committed:
Visual studio is crazy and did not fixate one change

WittleWolfie committed:
Basic cleanup

- Unify publicize task
- Remove imports

WittleWolfie committed:
Rename

WittleWolfie committed:
Resolve ambiguity error

WittleWolfie committed:
Update UISettingsEntityDropDownModMenuEntry.cs

WittleWolfie committed:
Update UISettingsEntityDropDownModMenuEntry.cs

WittleWolfie committed:
Rename / cleanup

WittleWolfie committed:
Update UISettingsEntityDropDownModMenuEntry.cs

WittleWolfie committed:
More cleanup

WittleWolfie committed:
More cleanup

WittleWolfie committed:
More cleanup

WittleWolfie commented:
Pushed some fixes and a bunch of cleanup. A few general notes:

- Lower case for method args, caps for class / struct members
- Don't use single line if / else statements (always put the operation on the next line, indented--single line statements don't need braces)
- Don't make new exceptions unless there's a functional need--generally ArgumentException and InvalidOperationException are sufficient w/ the message

ScaredKurufinve committed:
All sorts of changes

ScaredKurufinve committed:
Added a safety check for Localization Manager initialized state to avoid any possible NRE when accessing CurrentPack during when calling CreateString. strings will be registered later anyway when the first pack is loaded.
Added  another check for a <null>-returned value from Localized string in the Register method (just in case). And just generally formatted it more to my liking. Sorry :(

ScaredKurufinve committed:
Fixed the constructor of SettingsEntityModMenuEntry to be properly private, because it is meant to be used only internally for creation of the static instance.

ScaredKurufinve committed:
Made it so that former Collapsible Headers VMs return regular Header Views for the sake of backward compatibility (their place now is taken by former Collapsible Subheaders).
Also made a couple of Default Button transpilers prettier.

ScaredKurufinve committed:
Added a missing empty tooltip description to the Empty instance (so that the description view not show <null>). Also moved the creation of the instance inside the actual static constructor to make it prettier. (Is it?)

ScaredKurufinve committed:
Fixed a horrible bug where the builder would not clean the settings buffer after a new settings group has been declared.

ScaredKurufinve committed:
deleted a couple random usings (Vs be VS)

ScaredKurufinve committed:
Fixed a stupid bug with the order of ModEntry constructor calls causing only the first settings group passed to the constructor was actually stored.

ScaredKurufinve committed:
Decided to make ModsMenuEntry internal. Removed all its constructors' overloads and added several overloads of ModMenu.AddSettings in their stead. Extracted ModInfo structure from inside ModsMenuEntry.
Added Chinese, German and French localization.
Changed the construction of static UISettingsEntityDropdownModMenuEntry instance into unity's ScriptableObject CreateInstance to remove the warning in the log.

WittleWolfie committed:
Merge branch 'main' into main
